docs/DeveloperGuide.adoc

```diff
@@ -153,87 +154,230 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+
+image::EditSequenceDiagram.png[]
 
 ==== Design Considerations
 
-===== Aspect: How undo & redo executes
+===== Aspect: How the replacement Card is formed during the start of execution
+
+* **Alternative 1 (current choice):** A blank string in a face of the new Card means we use the face value in the Card to be replaced.
+** Pros: Easy to implement. Can use the extracted values in the arguments as is. Let the final step (`UniqueCardList#replace()`) handle the replacement logic.
+** Cons: From `EditCardCommandParser` until the end of the command execution in `LogicManager`, there may exist a Card with a face containing a blank string. May not be a desirable object to have.
+* **Alternative 2:** Get the Card to be edited directly in `EditCardCommandParser` so we can immediately produce the new Card with its final face values.
+** Pros: The Card to replace the old Card will be fully formed from the start.
+** Cons: No separation of parsing and model management logic since we would need to do a view check and get a Card from the current Deck all while in the parser.
+// end::editcard[]
+
+// tag::play[]
+=== Play mode: create a new session with a specific deck
+==== Current Implementation
+
+The `play` command creates a new session to play with a specific deck. The format of the `play` command:
+`play INDEX`
+
+The play command changes the mode of the application to `PLAY` mode and creates a new session with the Deck at the given `INDEX`.
+The value of the `FRONT` of the selected `Deck` will be displayed to the user.
+
+The first validation of the `play` command is performed in `PlayCommandParser#parse()`.
+The validation only checks that the `play` command has the correct format as the `INDEX` argument is given by the user and it is performed on the login level.
+
+In `PlayCommandParser#parse()`, the `INDEX` of the deck is extracted from the arguments in the `play` command. The `INDEX` is converted to an Index object. An `PlayCommand`
+object is then constructed with the Index.
+
+After the object of the `PlayCommand' is constructed, `PlayCommand#execute()` will be executed and the second validation of the `play` command is performed.
+This validation firstly checks if the given `INDEX` argument is a non-negative integer and is within the number of cards in the selected Deck. Then the validation checks if
+there is any card currently in the selected deck by checking if the `FRONT` face and `BACK` face of the card returned by `ModelManager#play()` are both empty.
+
+A valid `play` command will change the `MODE` of the `ModelManager` to `PLAY` mode and an `GameManager` object will be constructed in `ModelManager`.
+The first card of the selected deck is obtained using `deck#asUnmodifiableObservableList().get(0)` and returned to UI. The `FRONT` face of the first card will be displayed to the user.
+
+The following sequence diagram shows how the `play` operation works.
+
+image::PlaySequenceDiagram.png[]
```

Comment:
This sequence diagram and the subsequent sequence diagrams refer to their returned Command sub objects as "e". We think it should correspond to the first alphabet of the name of the subclass. 

Also the rightmost "e" under the Model box is a little ambiguous as what it refers to. Perhaps the GameManager instance. Regardless please reflect it on the next iteration of the diagram.

Also it is unclear as to where commandResult is instantiated. Consider showing where it is being in constructed.

For your consideration. Cheers!